src/machine/flash.go

```diff
@@ -91,21 +95,22 @@ func (fl *FlashBuffer) Write(p []byte) (n int, err error) {
 	// NOTE probably will not work as expected if you try to write over page boundary
 	// of pages with different sizes.
 	pagesize := uintptr(fl.b.EraseBlockSize())
-	currentPageCount := (fl.current - fl.start + pagesize - 1) / pagesize
-	totalPagesNeeded := (fl.current - fl.start + uintptr(len(p)) + pagesize - 1) / pagesize
-	if currentPageCount == totalPagesNeeded {
-		// just write the data
-		n, err := fl.b.WriteAt(p, int64(fl.current))
-		if err != nil {
-			return 0, err
-		}
-		fl.current += uintptr(n)
-		return n, nil
+
+	// calculate currentPageBlock relative to fl.start, meaning that
+	// block 0 -> fl.start
+	// block 1 -> fl.start + pagesize
+	// block 2 -> fl.start + pagesize*2
+	// ...
+	currentPageBlock := (fl.start + fl.offset - FlashDataStart()) / pagesize
+	lastPageBlockNeeded := (fl.start + fl.offset + uintptr(len(p)) - FlashDataStart()) / pagesize
```

Comment:
This no longer works for partial page writes, old logic contained `+ pagesize - 1` to allow for partial page writes.

Does it need documenting that len(p) must be a multiple of `EraseBlockSize()`?   (or restore the old logic)

Reply:
You are correct, I am modifying my code to restore that capability.

Reply:
Done.